tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateProduct/UpdateOptionsFeatureContext.php

```diff
@@ -0,0 +1,252 @@
+    {
+        $this->updateProductOptions($productReference, $table, ShopConstraint::shop($this->getDefaultShopId()));
+    }
+
+    /**
+     * @When I update product :productReference options for shop ":shopReference" with following values:
+     *
+     * @param string $productReference
+     * @param string $shopReference
+     * @param TableNode $table
+     */
+    public function updateProductOptionsForShop(string $productReference, string $shopReference, TableNode $table): void
+    {
+        $shopId = $this->getSharedStorage()->get(trim($shopReference));
+        $this->updateProductOptions($productReference, $table, ShopConstraint::shop($shopId));
+    }
+
+    /**
+     * @When I update product :productReference options for all shops with following values:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function updateProductOptionsForAllShops(string $productReference, TableNode $table): void
+    {
+        $this->updateProductOptions($productReference, $table, ShopConstraint::allShops());
+    }
+
+    /**
+     * @When I assign non existing manufacturer to product :productReference
+     *
+     * @param string $productReference
+     */
+    public function updateOptionsWithNonExistingManufacturer(string $productReference): void
+    {
+        // intentional. Mimics id of non-existing manufacturer
+        $nonExistingId = 50000;
+
+        try {
+            $command = new UpdateProductOptionsCommand(
+                $this->getSharedStorage()->get($productReference),
+                ShopConstraint::shop($this->getDefaultShopId())
+            );
+            $command->setManufacturerId($nonExistingId);
+            $this->getCommandBus()->handle($command);
+        } catch (ManufacturerException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Transform table:product option,value
+     *
+     * @param TableNode $tableNode
+     *
+     * @return ProductOptions
+     */
+    public function transformOptions(TableNode $tableNode): ProductOptions
+    {
+        $dataRows = $tableNode->getRowsHash();
+
+        return new ProductOptions(
+            $dataRows['visibility'],
+            PrimitiveUtils::castStringBooleanIntoBoolean($dataRows['available_for_order']),
+            PrimitiveUtils::castStringBooleanIntoBoolean($dataRows['online_only']),
+            PrimitiveUtils::castStringBooleanIntoBoolean($dataRows['show_price']),
+            $dataRows['condition'],
+            PrimitiveUtils::castStringBooleanIntoBoolean($dataRows['show_condition']),
+            $this->getManufacturerId($dataRows['manufacturer'])
+        );
+    }
+
+    /**
+     * @Then product :productReference should have following options:
+     *
+     * @param string $productReference
+     * @param ProductOptions $expectedOptions
+     */
+    public function assertOptionsForDefaultShop(string $productReference, ProductOptions $expectedOptions): void
```

Comment:
I don't think you need to copy the whole assertion methods, instead this context (or a single context made for the UpdateCommand only for write methods) could simply extend the initial one and override only the write methods

When everything is migrated, if the POC is validated, we can simply remove all the write methods related to little commands from the initial contexts which will then only serve as assertion contexts